### PR TITLE
Fix annotation changes for object with no annotations

### PR DESCRIPTION
**Pull request by roironn**
Fixes bug https://salto-io.atlassian.net/browse/SALTO-742.
The bug chronology:
1. When deploying a custom object, some annotation types are added by the adapter
2. Changes from the deploy return without a path hint
3. In update nacl files - the getLocations method fails since:
    a. There is no path hint
    b. the parent id of the annotation change - the annotation block is not present in any nacl file.
4. As a result, the groupAnnotationChanges function which would have fixed the change id to be the annotation block is never called, and the change is not applied. 

To fix this issue:
1. Modified the code in update nacl files to create the merged source map based on the change ids, and not on the ids of the changes created in the getUpdateChange method. This is OK since the method dose not modify the topLevelParent of the change ids, and since the getElementNaclFiles is using the top level ids. (so the merged source map is not changed)
2. Change the groupAnnotationChanges method to work with DetailedChanges and not DetailedChangesWithLocation (needed for #3)
3. Move the groupAnnotationChanges to be called from within getUpdateChanges method. This achieves two goals:
    a. The annotation changes are created *before* the getChangeLocation method is called, so the getChangeLocations works as expected and the change is not droped
    b. All of the modification to the update changes takes place in one location, which doesn't really affect the logic here, but seems to be a much more reasonable code organization.

### CODE REVIEW

File: packages/core/src/workspace/nacl_files/nacl_file_update.ts [hunk 1]
```diff
@@ -139,8 +139,7 @@ export const groupAnnotationTypeChanges = (fileChanges: DetailedChangeWithSource
         .value() },
     }
     // we should find a new location for the grouped change
-    return getChangeLocations(groupedChange, existingFileSourceMap
-      ?? {} as ReadonlyMap<string, SourceRange[]>)[0]
+    return groupedChange
```
Comment (ori-moisis) [inline packages/core/src/workspace/nacl_files/nacl_file_update.ts:142]: comment doesn't make sense anymore

File: packages/core/src/workspace/nacl_files/nacl_file_update.ts [hunk 2]
```diff
@@ -261,21 +260,22 @@ const wrapAdditions = (nestedAdditions: DetailedAddition[]): DetailedAddition =>
 
 const parentElementExistsInPath = (
   dc: DetailedChange,
-  elementsIndex: Record<string, string[]>
+  sourceMap: SourceMap
 ): boolean => {
   const { parent } = dc.id.createTopLevelParentID()
-  return (elementsIndex[parent.getFullName()] || [])
-    .includes(createFileNameFromPath(dc.path))
+  return !_.isEmpty(sourceMap.get(parent.getFullName())?.filter(
```
Comment (ori-moisis) [inline packages/core/src/workspace/nacl_files/nacl_file_update.ts:266]: is `!_.isEmpty` + `.filter` just a really elaborate way of implementing `_.some` or is this a different condition that I misunderstand?

File: packages/core/src/workspace/nacl_files/nacl_file_update.ts [hunk 3]
```diff
@@ -261,21 +260,22 @@ const wrapAdditions = (nestedAdditions: DetailedAddition[]): DetailedAddition =>
...
+    range => range.filename.includes(createFileNameFromPath(dc.path))
```
Comment (ori-moisis) [inline packages/core/src/workspace/nacl_files/nacl_file_update.ts:267]: is there a reason why the path won't match exactly and it needs to be `includes` here?

maybe this was just supposed to be
```
(sourceMap.get(parent.getFullName()) ?? [])
  .map(range => range.filename)
  .includes(createFileNameFromPath(dc.path))
```
like the original code?

File: packages/core/src/workspace/nacl_files/nacl_file_update.ts (hunk 2 above)
Comment (roironn) [reply packages/core/src/workspace/nacl_files/nacl_file_update.ts:266]: The only condition you missed is that I didn't think of some from some reason :)
Will change.

File: packages/core/src/workspace/nacl_files/nacl_file_update.ts (hunk 1 above)
Comment (roironn) [reply packages/core/src/workspace/nacl_files/nacl_file_update.ts:142]: 👍

**Pull request merged**